apps/extension/src/Approvals/ApproveBond/ConfirmLedgerBond.tsx

```diff
@@ -0,0 +1,181 @@
+import { useCallback, useState } from "react";
+import { toBase64 } from "@cosmjs/encoding";
+import BigNumber from "bignumber.js";
+
+import { LedgerError } from "@anoma/ledger-namada";
+import { Button, ButtonVariant } from "@anoma/components";
+import { defaultChainId as chainId } from "@anoma/chains";
+
+import { Ledger } from "background/ledger";
+import {
+  GetBondBytesMsg,
+  GetRevealPKBytesMsg,
+  SubmitSignedBondMsg,
+  SubmitSignedRevealPKMsg,
+} from "background/ledger/messages";
+import { Ports } from "router";
+import { closeCurrentTab } from "utils";
+import { useRequester } from "hooks/useRequester";
+import { Status } from "Approvals/Approvals";
+import {
+  ApprovalContainer,
+  ButtonContainer,
+} from "Approvals/Approvals.components";
+import { InfoHeader, InfoLoader } from "Approvals/Approvals.components";
+import { Message, RevealPKProps, Tokens } from "@anoma/types";
+import { SubmitRevealPKMsgSchema, RevealPKMsgValue } from "@anoma/types";
+import { QueryPublicKeyMsg } from "background/keyring";
+
+type Props = {
+  address: string;
+  msgId: string;
+  publicKey: string;
+};
+
+export const ConfirmLedgerBond: React.FC<Props> = ({
+  address,
+  msgId,
+  publicKey,
+}) => {
+  const requester = useRequester();
+  const [error, setError] = useState<string>();
+  const [status, setStatus] = useState<Status>();
+  const [statusInfo, setStatusInfo] = useState("");
+
+  const revealPk = async (ledger: Ledger, publicKey: string): Promise<void> => {
```

Comment:
NOTE: In my next PR, I will make this a top-level function for _any_ Ledger tx, as it applies to all tx signed with the Ledger (will refactor `Approvals` a bit to make this possible, and handle the different tx types more elegantly)